controller/state/statedb.go

```diff
@@ -124,7 +127,7 @@ func migrateDatabase(dbName string, dbInstance database.Driver) error {
 }
 
 // NewStateDB creates a state instance with a given driver and identity
-func NewStateDB(ctx context.Context, driver, conn string, identity crypto.PrivKey, recorder metrics.MetricsRecorder) (State, error) {
+func NewStateDB(ctx context.Context, driver, conn string, identity crypto.PrivKey, recorder metrics.MetricsRecorder, runNotice chan string) (State, error) {
```

Comment:
Nit: when adding parameters that are only used in test, I prefer having a second constructor function that has the additional params, and having the first constructor call the second with a nil value -- that way you don't have the extra param in regular non-test code (see controller/controller.go).

This is assuming you don't see a use for run notice outside of test.

Reply:
Yeah, I was considering that this could be useful functionality... but since I do not have a specific use for it yet, probably better to hide it behind a test-only constructor.